### PR TITLE
fix: sorting schema options alphabetically

### LINKED ISSUE #29059: [Task]: make schema sorting alphabetically on the basis of full name not just table name
### Is there an existing issue for this?

- [X] I have searched the existing issues

### SubTasks

https://www.notion.so/appsmith/5c8aa998aa94429995a6e0825ca2df05?v=14883d075b6649be9a8748938dcc224b&p=44a37bbc2e51478284f0e3a63846d0ef&pm=s

### PR DESCRIPTION
**Pull request by AmanAgarwal041**
### Previously
We were using `LinkedHashMap` to create a schema list which uses `insertion-order` to order the elements. 
Thereby creating a list of schema tables where the tables were alphabetically sorted first and then their names were appended with schema name as prefix which followed the order of tables instead of `schema.tableName` as a whole.

### Now
We have replaced the `LinkedHashMap` with `TreeMap` which uses `natural ordering` to order the elements.
Thereby creating a list of schema tables sorted by full name i.e. `schema.tableName`.

#### PR fixes following issue(s)
Fixes #29059
> if no issue exists, please create an issue and ask the maintainers about this first
>
>
- Bug fix (non-breaking change which fixes an issue)
- New feature (non-breaking change which adds functionality)
- Breaking change (fix or feature that would cause existing functionality to not work as expected)
- Chore (housekeeping or task changes that don't impact user perception)
- This change requires a documentation update
>
>
>
- [ ] Manual
- [ ] JUnit
- [ ] Jest
- [ ] Cypress
>
>
#### Dev activity
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

#### QA activity:
- [ ] [Speedbreak features](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#speedbreakers-) have been covered
- [ ] Test plan covers all impacted features and [areas of interest](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#areas-of-interest-)
- [ ] Test plan has been peer reviewed by project stakeholders and other QA members
- [ ] Manually tested functionality on DP
- [ ] We had an implementation alignment call with stakeholders post QA Round 2
- [ ] Cypress test cases have been added and approved by SDET/manual QA
- [ ] Added `Test Plan Approved` label after Cypress tests were reviewed
- [ ] Added `Test Plan Approved` label after JUnit tests were reviewed

## Summary by CodeRabbit

- **New Features**
  - Enhanced database schema sorting for MySQL and Postgres plugins to be case-insensitive.

- **Bug Fixes**
  - Ensured consistent case-insensitive ordering of table names in the schema for MySQL and Postgres data sources.

- **Tests**
  - Added regression tests to verify the correct sorting of schema in MySQL and Postgres data sources.

### CODE REVIEW

**Comment by NilanshBansal** [comment]
@AmanAgarwal041 are we doing sorting only for postgres and MySQL plugins?

**Review by NilanshBansal — DISMISSED** [review]

File: app/client/cypress/e2e/Regression/ClientSide/BugTests/SortedSchema_Spec.ts [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+  dataSources,
+} from "../../../../support/Objects/ObjectsCore";
+
+describe("Datasource structure schema should be sorted", () => {
+  it("1. Postgres schema should be in a sorted order", () => {
+    dataSources.CreateDataSource("Postgres");
+    assertHelper
+      .WaitForNetworkCall(`@getDatasourceStructure`)
+      .then(async (response) => {
+        const tables: any[] = response?.body.data?.tables || [];
```
[the remark this answers is not in the document]
Comment (AmanAgarwal041) [reply app/client/cypress/e2e/Regression/ClientSide/BugTests/SortedSchema_Spec.ts]: fixed

File: app/client/cypress/e2e/Regression/ClientSide/BugTests/SortedSchema_Spec.ts [hunk 2]
```diff
@@ -0,0 +1,50 @@
+import {
+  assertHelper,
+  dataSources,
+} from "../../../../support/Objects/ObjectsCore";
+
+describe("Datasource structure schema should be sorted", () => {
+  it("1. Postgres schema should be in a sorted order", () => {
```
[the remark this answers is not in the document]
Comment (AmanAgarwal041) [reply app/client/cypress/e2e/Regression/ClientSide/BugTests/SortedSchema_Spec.ts:7]: fixed

File: app/client/cypress/e2e/Regression/ClientSide/BugTests/SortedSchema_Spec.ts [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+    dataSources.CreateDataSource("Postgres");
+    assertHelper
+      .WaitForNetworkCall(`@getDatasourceStructure`)
+      .then(async (response) => {
```
[the remark this answers is not in the document]
Comment (AmanAgarwal041) [reply app/client/cypress/e2e/Regression/ClientSide/BugTests/SortedSchema_Spec.ts]: removed

File: app/client/cypress/e2e/Regression/ClientSide/BugTests/SortedSchema_Spec.ts [hunk 4]
```diff
@@ -0,0 +1,49 @@
+import {
+  assertHelper,
+  dataSources,
+} from "../../../../support/Objects/ObjectsCore";
+
+describe("Datasource structure schema should be sorted", () => {
+  it("1. Postgres schema should be in a sorted order", () => {
```
Comment (Aishwarya-U-R) [inline app/client/cypress/e2e/Regression/ClientSide/BugTests/SortedSchema_Spec.ts:7]: Please add BugId to `it` block.

File: app/client/cypress/e2e/Regression/ClientSide/BugTests/SortedSchema_Spec.ts [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+    dataSources.CreateDataSource("Postgres");
```
Comment (Aishwarya-U-R) [inline app/client/cypress/e2e/Regression/ClientSide/BugTests/SortedSchema_Spec.ts:8]: When only one table is created, how are we verifying its sorted?

Comment (AmanAgarwal041) [reply app/client/cypress/e2e/Regression/ClientSide/BugTests/SortedSchema_Spec.ts:8]: Its handled in the sorted check logic : https://github.com/appsmithorg/appsmith/pull/29460/files#diff-a0af5891d56e9858603baf582345e33956838a09e53702b27b42d26df180f855R15

**Review by Aishwarya-U-R — DISMISSED** [review]
Cypress looks good.

File: app/server/appsmith-plugins/postgresPlugin/src/test/java/com/external/plugins/PostgresPluginTest.java [hunk 6]
```diff
@@ -486,9 +486,15 @@ public void testStructure() {
...
+                    for (DatasourceStructure.Table table : structure.getTables()) {
+                        if (table.getName().equals(campusTableToBeFound)) {
+                            campusTable = table;
+                            break;
+                        }
+                    }
+                    assertNotNull(campusTable);
                     assertEquals(DatasourceStructure.TableType.TABLE, campusTable.getType());
                     assertArrayEquals(
                             new DatasourceStructure.Column[] {
```
[the remark this answers is not in the document]
Comment (AmanAgarwal041) [reply app/server/appsmith-plugins/postgresPlugin/src/test/java/com/external/plugins/PostgresPluginTest.java]: Done

**Review by NilanshBansal — APPROVED** [review]

**Pull request merged**